### ISSUE TITLE
Nothing visible in variable-inspector for jupyterlab(windows)

### ISSUE THREAD

**Issue by AndeYashwanth** [lckr/jupyterlab-variableInspector#130]
When I run ```jupyter notebook``` I get the variable inspector floating box(display at startup) which is working fine.
But when I run ```jupyter lab``` and ```rightclick > open variable inspector``` I get variable inspector tab with column names [name,type...]. 
When I run a cell nothing is being displayed in the tab. This issue is only when I use jupyterlab(not jupyter notebook).

My jupyter lab version
```
C:\Users\andey>pip freeze | grep jupy
jupyter==1.0.0
jupyter-client==6.0.0
jupyter-console==6.1.0
jupyter-contrib-core==0.3.3
jupyter-contrib-nbextensions==0.5.1
jupyter-core==4.6.3
jupyter-highlight-selected-word==0.2.0
jupyter-latex-envs==1.4.6
jupyter-nbextensions-configurator==0.4.1
jupyter-server==0.1.1
jupyter-server-proxy==1.2.0
jupyterlab==2.0.1
jupyterlab-code-formatter==1.2.0
jupyterlab-github==2.0.0
jupyterlab-pygments==0.1.0
jupyterlab-server==1.0.7
jupyterthemes==0.20.0
```

Installed extension using
```
pip install ipywidgets
jupyter labextension install @jupyter-widgets/jupyterlab-manager
jupyter labextension install @lckr/jupyterlab_variableinspector
``` 

jupyterlab extension list
```
C:\Users\andey>jupyter labextension list
JupyterLab v2.0.1
Known labextensions:
   app dir: c:\users\andey\appdata\local\programs\python\python37\share\jupyter\lab
        @jupyter-widgets/jupyterlab-manager v2.0.0 enabled  ok
        @jupyterlab/celltags v2.0.2 enabled  ok
        @jupyterlab/debugger v0.2.0-beta.1 enabled  ok
        @jupyterlab/github v2.0.0 enabled  ok
        @jupyterlab/toc v3.0.0 enabled  ok
        @lckr/jupyterlab_variableinspector v0.4.0 enabled  ok
        @ryantam626/jupyterlab_code_formatter v1.2.0 enabled  ok
```

**Comment by zhangyonglzlm** [comment]
I am faced with the same issue on CentOS.

**Comment by martintb** [comment]
Same issue on macOS. I'm wondering if it's a JupyterLab v1.0 vs v2.0v compatibility problem.

**Comment by Leo1215** [comment]
same issue on Arch Linux with Jupyterlab 2.0

**Comment by martintb** [comment]
So this is definitely a JupyterLab v2.0 problem. See #129 where the package author confirms and says that they're working on an update.

**Comment by lckr** [comment]
An updated version compatible with Jupyterlab 2.0 has now been published to github and npm.
If there are any remaining issues, please let me know.

**Comment by linuxcf** [comment]
After this issue was closed, I am facing the same problem on conda on Linux (stock Ubuntu 16.04).  jupyter lab is absolutely up-to-date as of this posting (ver. 2.0.1), updated via conda and if I remember correctly, I have seen other jupyter server, jupyter client, npm... updates.  But variableinspector does nothing; it just sits there, regardless of being asked to attach to the correct IPython kernel.

**Comment by lckr** [comment]
Hey @linuxcf, can you check if the variable inspector extension pushed its custom code to the kernel (then this has to be a display problem). You can do this by checking if the function `_jupyterlab_variableinspector_dict_list()` is available to the kernel.

**Comment by byoung4** [comment]
> Hey @linuxcf, can you check if the variable inspector extension pushed its custom code to the kernel (then this has to be a display problem). You can do this by checking if the function `_jupyterlab_variableinspector_dict_list()` is available to the kernel.

```
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
In  [2]:
Line 1:     _jupyterlab_variableinspector_dict_list()

NameError: name '_jupyterlab_variableinspector_dict_list' is not defined
---------------------------------------------------------------------------
```

It appears that I only have the following `variableinspector` functions available.

```
_jupyterlab_variableinspector_Jupyter
_jupyterlab_variableinspector_nms
```

I am running a fresh install of JupyterLab 2.1.1 with python 3.8.2.

**Comment by linuxcf** [comment]
@lckr : Running '_jupyterlab_variableinspector_dict_list()' in a new cell in an active jupyterlab window, I get the error message you mention above:
```
NameError                                 Traceback (most recent call last)
<ipython-input-11-a2d401806d89> in <module>
----> 1 _jupyterlab_variableinspector_dict_list()

NameError: name '_jupyterlab_variableinspector_dict_list' is not defined

```

**Comment by lckr** [comment]
Ok thanks, I tried to reproduce this on multiple machines for `jupyterlab 2.1.1` installed via conda-forge and the variable extension installed via `jupyter labextension install @lckr/jupyterlab_variableinspector` (therefore resulting in version 0.5) . In the end I wasn't able to reproduce this issue yet.

What version of the extension are you using?

**Comment by byoung4** [comment]
Alright. I wasn't getting any positive results in a new Conda environment, so I deleted my *entire* Conda environment (`rm -rf ~/miniconda3`) and started over from scratch.

For some reason, it now works. Specifically, it works on Python 3.8 when the only things installed are Jupyter Lab and @lckr/jupyterlab_variableinspector, and it also works with all the rest of my environment installed.

It *does* still demonstrate the previous behavior if I change my Notebook kernel from `Python 3` to `xpython` (i.e., [xeus-python](https://github.com/jupyter-xeus/xeus-python), for use with the [Jupyter Debugger](https://github.com/jupyterlab/debugger)). In that case, the only two objects exposed are

```
_jupyterlab_variableinspector_Jupyter
_jupyterlab_variableinspector_nms
```

**Comment by linuxcf** [comment]
@lckr:  I tried your recommendation.  In the process, I was told I also had to install node.js and npm before trying to install jupyterlab.  After all was said and done, **it worked!**  Thank you!  Now, I need to figure out why my base environment has not updated to Python 3.8...  Another odd tidbit of information:  Spyder was also segfaulting on startup, but in your suggested test environment it doesn't!  So, I need to reinstall my whole anaconda, as @byoung4 did.

**Comment by byoung4** [comment]
@linuxcf I'm using the latest version of Miniconda, which defaults to Python 3.7.

You can create a new environment with Python 3.8 by
```bash
conda create -n foo python=3.8
```
or by upgrading your base Conda, either through the official Anaconda channel
```bash
conda install -c anaconda python=3.8
```
or through conda-forge (this is what I do)
```bash
# use these commands while the base environment is activated
conda config --add channels conda-forge
conda config --set channel_priority strict 
conda update conda
conda update python
conda update --all
```

Also, you can either install nodejs+npm through your OS (`sudo apt install nodejs` or `brew install node`) or with Conda (`conda install nodejs`).

**Comment by linuxcf** [comment]
@byoung4: Thank you for your advice.  I actually totally erased my '~/miniconda3' directory, where anaconda was installed, downloaded the miniconda installer, and ran the shell script to start rebuilding anaconda.  Before I did anything else, I updated Python, and now it is 3.8.2.  Then I installed numpy, scipy, pandas, statsmodels, scikit-rf, scikit-learn,...   I installed node.js and npm thourgh anaconda.    Everything seems to be fine now, for the time being at least.  I don't understand why miniconda installs with Python 3.7 and why anaconda did not upgrade Python previously, keeping it at 3.7.  It had only been since February, when I had also erased and reinstalled anaconda.  I thought the anaconda environment  took care of updates and dependencies.  I did not run 'conda config --set channel-priority strict', so I don't know if that was my problem.  Again, thank you for your advice.
